src/test/coins_tests.cpp

```diff
@@ -881,6 +881,135 @@ BOOST_AUTO_TEST_CASE(coins_coinbase_spends)
     }
 }
 
+// This test is similar to the previous test
+// except the emphasis is on testing the functionality of UpdateCoins
+// random txs are created and UpdateCoins is used to update the cache stack
+// In particular it is tested that spending a duplicate coinbase tx
+// has the expected effect (the other duplicate is overwitten at all cache levels)
```

Comment:
Simplify the test by removing the testing of duplicate coinbase transactions:

* delete the last sentence of the above comment
* delete the ``spent_a_duplicate_coinbase`` and ``duplicateids`` declarations
* delete " unless intended to duplicate" in the comment on line 910
* delete the "1/100 times create a duplicate coinbase" ``if`` body
* delete the comment that mentions BIP 34 (but not the code it refers to! just the comment is inapplicable)
* delete the comment starting "The test is designed" and the two code lines after it.